nntrainer/compiler/tflite_interpreter.cpp

```diff
@@ -422,12 +415,91 @@ buildOperatorCodes(const TfOpIdxMap &map, flatbuffers::FlatBufferBuilder &fbb) {
   return fbb.CreateVector(fb_op_codes);
 }
 
+/**
+ * @brief get Serialized Tensor shape
+ *
+ * @param dim Tensor dimension to convert
+ * @param type tflite Tensor type
+ * @return flatbuffers::Offset<flatbuffers::Vector<int32_t>>
+ */
+flatbuffers::Offset<flatbuffers::Vector<int32_t>>
+buildTesorShape(const TensorDim &dim, flatbuffers::FlatBufferBuilder &fbb,
+                int effective_dimension, bool is_signature = false) {
+  std::vector<int32_t> fb_dimension;
+
+  /// batch dimension is always considered with signature
+  if ((effective_dimension >> 3) & 1) {
+    fb_dimension.push_back(is_signature ? -1 : dim.batch());
+  }
+
+  for (unsigned int i = 1; i < MAXDIM; ++i) {
+    if (((effective_dimension >> (MAXDIM - i - 1)) & 1) == 0) {
+      continue;
+    }
+    fb_dimension.push_back(dim.getTensorDim(i));
+  }
+
+  return fbb.CreateVector(fb_dimension);
+}
+
+flatbuffers::Offset<flatbuffers::Vector<flatbuffers::Offset<tflite::Tensor>>>
+buildTensors(const TfOpIdxMap &map, flatbuffers::FlatBufferBuilder &fbb) {
+  /// @todo: the actual (suqeezed) tensor dimension must be known before coming
+  /// here. For now, it is directly guessed for the fc layer
+  const auto &variables = map.getIndexMap<const Var_Grad *>().getData();
+  const auto &buffer_map = map.getIndexMap<const float *, TfOpIdxMap::Buffer>();
+
+  std::vector<flatbuffers::Offset<tflite::Tensor>> fb_tensors;
+  fb_tensors.reserve(variables.size());
+
+  auto create_tensor = [&fbb, &buffer_map](const Var_Grad *var) {
+    /// @fixme: There is a fixed assumption that effective_dimension is 0b1001,
+    /// this is not true, this should be supported inherently from TensorDim
+    /// class.
+    int FIX_ME_FLAG = 0b1001;
+    auto shape = buildTesorShape(var->getDim(), fbb, FIX_ME_FLAG, false);
+    auto shape_sig = buildTesorShape(var->getDim(), fbb, FIX_ME_FLAG, true);
+    auto name = fbb.CreateString(var->getName());
+    auto tensor = var->getVariableRef();
+
+    unsigned int buffer_idx = 1;
+    if (!tensor.uninitialized() && tensor.isAllocated()) {
```

Comment:
in case, the tensor is not initialized, what difference would it make to the generated output file? this question is not for the weights but for the input/output tensors perspective.

Reply:
it will point to a empty buffer (in our case, index of 1). `tflite::interpreter` will regard this as a tensor that needs input from user.

Reply:
so, if a model is saved to tflite format before allocation input/output buffers, tflite assumes that all the input/output buffers must be given by the user?

Reply:
For that reason, I made it not to save buffer(as far as I remember) for in/out, so that it can be inputted from user.

Reply:
> For that reason, I made it not to save buffer(as far as I remember) for in/out, so that it can be inputted from user.

got it.

> so, if a model is saved to tflite format before allocation input/output buffers, tflite assumes that all the input/output buffers must be given by the user?

I am talking the input/output buffers of the intermediate layers.

Reply:
Ah, right. this will is not required to be filled from user. the operations fill it.

Reply:
So, empty buffers for model inputs/outputs are treated differently than empty buffers per operation inputs/outputs, right?

Reply:
In terms of list of tensors it is treated the same. `SubGraph` table defines which index is model in and out.

Reply:
great, thanks for the explanation.